internal/pkg/cli/local_run.go

```diff
@@ -106,11 +198,205 @@ func (o *localRunOpts) Execute() error {
 	}
 
 	secrets := taskDef.Secrets()
-	_, err = o.ecsLocalClient.DecryptedSecrets(secrets)
+	decryptedSecrets, err := o.ecsLocalClient.DecryptedSecrets(secrets)
 	if err != nil {
 		return fmt.Errorf("get secret values: %w", err)
 	}
 
+	secretsList := make(map[string]string)
+	for _, s := range decryptedSecrets {
+		secretsList[s.Name] = s.Value
+	}
+
+	envVars := make(map[string]string)
+	envVariables := taskDef.EnvironmentVariables()
+	for _, envVariable := range envVariables {
+		envVars[envVariable.Name] = envVariable.Value
+	}
+
+	containerPorts := make(map[string]string)
+	containerdef := taskDef.ContainerDefinitions
+	for _, container := range containerdef {
+		for _, portMapping := range container.PortMappings {
+			hostPort := aws.Int64Value(portMapping.HostPort)
+			hostPortStr := strconv.FormatInt(hostPort, 10)
+			var containerPort int64
+			if portMapping.ContainerPort == nil {
+				containerPort = hostPort
+			} else {
+				containerPort = aws.Int64Value(portMapping.ContainerPort)
+			}
+			containerportStr := strconv.FormatInt(containerPort, 10)
+			containerPorts[hostPortStr] = containerportStr
+		}
+	}
+
+	env, err := o.store.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("get environment %q configuration: %w", o.envName, err)
+	}
+	o.targetEnv = env
+
+	app, err := o.store.GetApplication(o.appName)
+	if err != nil {
+		return fmt.Errorf("get application %q configuration: %w", o.appName, err)
+	}
+	o.targetApp = app
+
+	envSess, err := o.sessProvider.FromRole(env.ManagerRoleARN, env.Region)
+	if err != nil {
+		return fmt.Errorf("get env session: %w", err)
+	}
+	mft, err := workloadManifest(&workloadManifestInput{
+		name:         o.wkldName,
+		appName:      o.appName,
+		envName:      o.envName,
+		interpolator: o.newInterpolator(o.appName, o.envName),
+		ws:           o.ws,
+		unmarshal:    o.unmarshal,
+		sess:         envSess,
+	})
+	if err != nil {
+		return err
+	}
+	o.appliedDynamicMft = mft
+	o.repository, err = o.configureClients(o)
+	if err != nil {
+		return err
+	}
+
+	err = o.buildContainerImages(o)
+	if err != nil {
+		return fmt.Errorf("building container image: %w", err)
+	}
+
+	containerNames := o.out.ContainerNames
+	imageNames := o.out.ImageNames
+
+	for i, image := range imageNames {
+		o.imageInfoList = append(o.imageInfoList, imageInfo{
+			containerName: containerNames[i],
+			imageURI:      image,
+		})
+	}
+
+	var sideCarListInfo []imageInfo
+	manifestContent := o.appliedDynamicMft.Manifest()
+	switch t := manifestContent.(type) {
+	case *manifest.ScheduledJob:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.LoadBalancedWebService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.WorkerService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.BackendService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	}
+	o.imageInfoList = append(o.imageInfoList, sideCarListInfo...)
```

Comment:
After reading the code, I understand now that `sideCarListInfo` contains all the sidecars _whose image are specified via. `image.location`_. The image information of the sidecars that used `image.build` are _already_ in `o.imageInfoList` at this point.

However, this was not obvious. At first, it seemed to me that we were including the sidecar image info twice - once from `buildContainerImages`, and once from `getBuiltSideCarImages`.

I guess the easiest way right now to improve this is just to rename the variables and functions to `sidecarImageLocations`, and maybe add a comment here. I do have the intuition though that we should think about how to refactor these when time permits.